Library/Homebrew/utils.rb

```diff
@@ -404,6 +404,36 @@ def gzip(*paths)
     end
   end
 
+  def reproducible_gzip(path, mtime: ENV["SOURCE_DATE_EPOCH"].to_i, orig_name: File.basename(path),
```

Comment:
May even want to create a new `Utils::Gzip` or `Utils::Reproducible` just for this and call `Utils::Gzip.reproducible_gzip` as a more explicit API. In an ideal world we'd probably not rely on this `utils` global include.

If desired could also have a `reproducible_gzips` helper or something that does the looping, if it'll make the homebrew/core formulae usage nicer.